### PR TITLE
Practice 12

### PR DESCRIPTION
**Pull request by Myroslava-Movchan**
Мені не вдалося виправити помилку щодо кодування публічного ключа (System.FormatException: The input is not a valid Base-64 string as it contains a non-base 64 character, more than two padding characters, or an illegal character among the padding characters.). На жаль, я так і не знайшла проблему, тому не маю скріншоту, який можу прикріпити. 
Не встигла покрити 80% коду юніт тестами. Є покриття всіх сервісів.

### CODE REVIEW

File: Unit Tests/CustomerControllerTest.cs [hunk 1]
```diff
@@ -0,0 +1,91 @@
+﻿using Microsoft.AspNetCore.Mvc;
```
Comment (liudmyla-zinchenko) [inline Unit Tests/CustomerControllerTest.cs:1]: Avoid committing code coverage reports and db files with gitingnore.

File: Unit Tests/Unit Tests.csproj [hunk 2]
```diff
@@ -0,0 +1,31 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net8.0</TargetFramework>
+    <RootNamespace>Unit_Tests</RootNamespace>
```
Comment (liudmyla-zinchenko) [inline Unit Tests/Unit Tests.csproj]: It is better to call your project like Online_Store_Management.Unit_Tests
Following pattern {main_project}.{type_of tests}_Tests

**Review by liudmyla-zinchenko — COMMENTED** [review]
Comments are provided

**Comment by Myroslava-Movchan** [comment]
image
image
screenshots of working encryption and decryption @BraveheartMousy

File: Online Store Management/Infrastructure/RSAEncryption.cs [hunk 3]
```diff
@@ -13,27 +14,35 @@ public static (string PublicKey, string PrivateKey) GenerateKeys()
...
             );
         }
-        public static RSA ImportPublicKey(string pemKey)
+        public static RSA ImportPublicKey(string base64OrPemKey)
         {
-            var base64Key = string.Join("\n", pemKey
-                .Split('\n')
-                .Where(line => !line.StartsWith("-----")));
+            var keyContent = base64OrPemKey
+                .Replace("-----BEGIN PUBLIC KEY-----", string.Empty)
```
Comment (liudmyla-zinchenko) [inline Online Store Management/Infrastructure/RSAEncryption.cs]: So basically you do not need this piece if you have key itself as you set it is config file

**Review by liudmyla-zinchenko — COMMENTED** [review]
Please resolve the comments below

File: Online Store Management/Infrastructure/RSAEncryption.cs (hunk 3 above)
Comment (Myroslava-Movchan) [reply Online Store Management/Infrastructure/RSAEncryption.cs]: please check, fixed RSAEncryption, root namespace and updated .gitignore

**Review by lzinchenko-CLGX — APPROVED** [review]
Accepted

**Review by liudmyla-zinchenko — APPROVED** [review]
Accepted

**Pull request merged**